src/e2ee/matrixKeyProvider.ts

```diff
@@ -41,7 +41,9 @@ export class MatrixKeyProvider extends BaseKeyProvider {
       encryptionKeys,
     ] of this.rtcSession.getEncryptionKeys()) {
       for (const [index, encryptionKey] of encryptionKeys.entries()) {
```

Comment:
As it is currently written (and is sufficiently captured in type hints), the array returned by `getEncryptionKeys` is a [sparse array](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Indexed_collections#sparse_arrays), which contains empty slots that are skipped by array iteration methods.

Thus, instead of using a `for...of` loop and having to check for `undefined`, this could simply use `forEach`:

```suggestion
      encryptionKeys.forEach((encryptionKey, index) => {
```

Note that this will work only if the array returned by `getEncryptionKeys` remains typed as `Array<Uint8Array>` instead of `Array<Uint8Array | undefined>`.

Reply:
Well spotted. 👍 However, this is now redundant given the refactor from https://github.com/matrix-org/matrix-js-sdk/pull/4427